### PR TITLE
Batch #2 - 11. No .builder() API: No classes should have a .builder()

### PR DESCRIPTION
**Pull request by vhvb1989**
add MethodName rule to disable builder as a method name and update isssues

You might want to see checkstyle.xml for MethodName rule update.  Then everything else is fixing what we had already with a builder()

### CODE REVIEW

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -60,7 +60,12 @@ page at http://checkstyle.sourceforge.net/config.html -->
     <module name="LocalFinalVariableName"/>
     <module name="LocalVariableName"/>
     <module name="MemberName"/>
-    <module name="MethodName"/>
+    <module name="MethodName">
+      <!-- Use default MethodName but block the use of 'builder' as method name -->
+      <property name="format" value="(?=^[a-z][a-zA-Z0-9]*$)(?!^[b][uU][iI][lL][dD][eE][rR]$)"/>
```
Comment (conniey) [inline eng/code-quality-reports/src/main/resources/checkstyle/checkstyle.xml]: Are mode specifiers available? It's easier to read `((?i)builder)$

Comment (vhvb1989) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle.xml]: Yes, it is supported. I have tested and worked.
Thank you for it! 👍  Updated now

File: sdk/core/azure-core/src/main/java/com/azure/core/http/HttpPipelineBuilder.java [hunk 2]
```diff
@@ -41,7 +41,12 @@
     private List<HttpPipelinePolicy> pipelinePolicies;
 
 
-    HttpPipelineBuilder() {
+    /**
+     -     * Creates a builder that can configure options for the HttpPipeline before creating an instance of it.
+     -     *
+     -     * @return A new {@link HttpPipelineBuilder} to create a HttpPipeline from.
+     */
+    public HttpPipelineBuilder() {
```
Comment (JonathanGiles) [inline sdk/core/azure-core/src/main/java/com/azure/core/http/HttpPipelineBuilder.java]: Can you please fix up the javadoc here. Thanks.

Comment (vhvb1989) [reply sdk/core/azure-core/src/main/java/com/azure/core/http/HttpPipelineBuilder.java]: on it

**Pull request merged**